src/MainWindow.vala

```diff
@@ -33,49 +32,47 @@ public class Friends.MainWindow : Hdy.ApplicationWindow {
-            show_close_button = true
+        var sidebar_header = new Gtk.WindowHandle () {
+            child = new Gtk.WindowControls (Gtk.PackType.START)
         };
+        sidebar_header.add_css_class (Granite.STYLE_CLASS_DEFAULT_DECORATION);
+        sidebar_header.add_css_class ("titlebar");
+        sidebar_header.add_css_class (Granite.STYLE_CLASS_FLAT);
 
-        unowned Gtk.StyleContext sidebar_header_context = sidebar_header.get_style_context ();
-        sidebar_header_context.add_class ("default-decoration");
-        sidebar_header_context.add_class (Gtk.STYLE_CLASS_FLAT);
-
-        var individualview_header = new Hdy.HeaderBar () {
-            decoration_layout = ":maximize",
-            has_subtitle = false,
-            show_close_button = true
+        var individualview_header = new Gtk.WindowHandle () {
+            child = new Gtk.WindowControls (Gtk.PackType.END) {
+                halign = Gtk.Align.END
+            }
+        };
+        individualview_header.add_css_class (Granite.STYLE_CLASS_DEFAULT_DECORATION);
+        individualview_header.add_css_class ("titlebar");
+        individualview_header.add_css_class (Granite.STYLE_CLASS_FLAT);
+
+        search_entry = new Gtk.SearchEntry () {
+            margin_start = 9,
+            margin_end = 9,
+            margin_top = 3,
+            margin_bottom = 6,
+            hexpand = true,
+            placeholder_text = _("Search Friends"),
+            valign = Gtk.Align.CENTER
         };
 
-        unowned Gtk.StyleContext individualview_header_context = individualview_header.get_style_context ();
-        individualview_header_context.add_class ("default-decoration");
-        individualview_header_context.add_class (Gtk.STYLE_CLASS_FLAT);
-
-        search_entry = new Gtk.SearchEntry ();
-        search_entry.margin_start = search_entry.margin_end = 9;
-        search_entry.margin_top = 3;
-        search_entry.margin_bottom = 6;
-        search_entry.hexpand = true;
-        search_entry.placeholder_text = _("Search Friends");
-        search_entry.valign = Gtk.Align.CENTER;
-
-        listbox = new Gtk.ListBox ();
-        listbox.activate_on_single_click = true;
-        listbox.expand = true;
-        listbox.selection_mode = Gtk.SelectionMode.SINGLE;
+        listbox = new Gtk.ListBox () {
+            activate_on_single_click = true,
+            hexpand = true,
+            vexpand = true,
+            selection_mode = Gtk.SelectionMode.SINGLE,
+        };
         listbox.set_filter_func (filter_function);
         listbox.set_header_func (header_function);
         listbox.set_sort_func (sort_function);
 
-        var scrolledwindow = new Gtk.ScrolledWindow (null, null);
-        scrolledwindow.add (listbox);
+        var scrolledwindow = new Gtk.ScrolledWindow () {
+            child = listbox
+        };
 
         var sidebar_grid = new Gtk.Grid ();
-        sidebar_grid.get_style_context ().add_class (Gtk.STYLE_CLASS_SIDEBAR);
```

Comment:
If you remove this class then the list has a different background color than the other widgets in this pane. You might want to either leave in `Granite.STYLE_CLASS_SIDEBAR`, or use `background` on the ListBox and set `view` on the `individual_grid`. I'm a fan of the latter personally:

![Screenshot from 2022-06-02 14 56 00](https://user-images.githubusercontent.com/7277719/171745041-2b00d614-51b6-4812-b500-6c9db3328b61.png)

Reply:
Yes that's correct, there's a `Granite.STYLE_CLASS_VIEW`